apps/examples/testcase/le_tc/kernel/tc_pthread.c

```diff
@@ -1290,28 +1290,31 @@ static void tc_pthread_pthread_cond_signal_wait(void)
 	TC_ASSERT_EQ("pthread_mutex_lock", ret_chk, OK);
 
 	ret_chk = pthread_create(&pthread_waiter, NULL, thread_cond_signal, NULL);
-	TC_ASSERT_EQ("pthread_create", ret_chk, OK);
+	TC_ASSERT_EQ_CLEANUP("pthread_create", ret_chk, OK, pthread_mutex_unlock(&g_mutex); pthread_mutex_destroy(&g_mutex));
```

Comment:
To check unlock & destroy inside of TC_ASSERT like a line 1315~1316, how about below style?
```
TC_ASSERT_EQ_CLEANUP("pthread_mutex_lock", g_cond_sig_val, VAL_ONE, goto errout); 
...
TC_ASSERT_EQ_CLEANUP("pthread_cond_signal", ret_chk, OK, goto errout_with_lock);
...
errout_with_lock :
   pthread_mutex_unlock;
errout:
  ret_chk = pthread_mutex_dextroy(&g_mutex);

Reply:
applied your suggestion at #3831 . so that close this.